src/main/java/org/opennars/control/concept/ProcessAnticipation.java

```diff
@@ -152,7 +150,76 @@ public static void maintainDisappointedAnticipations(final Parameters narParamet
             concept.memory.emit(OutputHandler.DISAPPOINT.class,concept.getTerm());
         }
         for(Concept.AnticipationEntry entry : disappointed) {
-            concept.memory.inputTask(time, entry.negConfirmation, false);
+            final Term term = entry.negConfirmation.getTerm();
+
+            { // revise with negative evidence
+                TruthValue truthOfBeliefWithTerm = null;
+                {
+
+                    Set<Term> targets = new LinkedHashSet<>();
+                    {
+                        //add to all components, unless it doesn't have vars
+                        Map<Term, Integer> ret = CompoundTerm.replaceIntervals(term).countTermRecursively(null);
```

Comment:
doesn't make sense, a belief cannot be in a subterm, only in the concept of the term of the belief itself.